src/main/java/hudson/remoting/Engine.java

```diff
@@ -403,6 +463,101 @@ public static Engine current() {
+                        result.put("trustStoreProvider", System.getProperty("javax.net.ssl.trustStoreProvider", ""));
+                        result.put("trustStorePasswd", System.getProperty("javax.net.ssl.trustStorePassword", ""));
+                        return result;
+                    }
+                });
+        KeyStore keystore = null;
+
+        FileInputStream trustStoreStream = null;
+        try {
+            String trustStore = properties.get("trustStore");
+            if (!"NONE".equals(trustStore)) {
+                File trustStoreFile;
+                if (trustStore != null) {
+                    trustStoreFile = new File(trustStore);
+                    trustStoreStream = getFileInputStream(trustStoreFile);
+                } else {
+                    String javaHome = properties.get("javaHome");
+                    trustStoreFile = new File(
+                            javaHome + File.separator + "lib" + File.separator + "security" + File.separator
+                                    + "jssecacerts");
+                    if ((trustStoreStream = getFileInputStream(trustStoreFile)) == null) {
+                        trustStoreFile = new File(
+                                javaHome + File.separator + "lib" + File.separator + "security" + File.separator
+                                        + "cacerts");
+                        trustStoreStream = getFileInputStream(trustStoreFile);
+                    }
+                }
+
+                if (trustStoreStream != null) {
+                    trustStore = trustStoreFile.getPath();
+                } else {
+                    trustStore = "No File Available, using empty keystore.";
+                }
+            }
+
+            String trustStoreType = properties.get("trustStoreType");
+            String trustStoreProvider = properties.get("trustStoreProvider");
+            LOGGER.log(Level.FINE, "trustStore is: {0}", trustStore);
+            LOGGER.log(Level.FINE, "trustStore type is: {0}", trustStoreType);
+            LOGGER.log(Level.FINE, "trustStore provider is: {0}", trustStoreProvider);
+
+            if (trustStoreType.length() != 0) {
+                LOGGER.log(Level.FINE, "init truststore");
+
+                if (trustStoreProvider.length() == 0) {
+                    keystore = KeyStore.getInstance(trustStoreType);
+                } else {
+                    keystore = KeyStore.getInstance(trustStoreType, trustStoreProvider);
+                }
+
+                char[] trustStorePasswdChars = null;
+                String trustStorePasswd = properties.get("trustStorePasswd");
+                if (trustStorePasswd.length() != 0) {
+                    trustStorePasswdChars = trustStorePasswd.toCharArray();
+                }
+
+                keystore.load(trustStoreStream, trustStorePasswdChars);
+                if (trustStorePasswdChars != null) {
+                    for (int i = 0; i < trustStorePasswdChars.length; ++i) {
```

Comment:
I agree, it's reasonable from the security perspective (at least if JIT is not smart enough to optimize it)